### PR TITLE
add hooks section

### LINKED ISSUE #9: Hooks section
Let's think about what we want to show there.

As a first iteration, I'll vote for the `useScaffoldContractWrite` hook. A good example sending some args + value, with a button that does writeAsync onClick.

Question: Should we add a "view all hooks" link or some text like that links to the hooks sections in the docs? Maybe not a primary button, it can be just an underlined link.

image

### PR DESCRIPTION
**Pull request by technophile-04**
## Description

https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/389df1a6-7c8e-49a4-b512-ba79a67deda4

Question : 

1. Should we keep it on mobile or remove it?

2. Maybe move the hooks section below "Debug contract sections", or maybe below "Components" section but below will break zig-zag patter which we are following.

Also please feel free to criticize the code logic or suggest another approach 😅, would love to iterate/revamp the logic 🙌

Fixes #9

### CODE REVIEW

**Comment by carletex** [comment]
This is looking great @technophile-04 !! Thank you.

I'd make a couple of tweaks:

- As you suggested. Move it below "Components":
   - Change the Hook Section bg to the light one
   - Switch columns left->right / right->left to keep the zig zag pattern
 
- I'd like the hook section text to mention autocompletion. Something like "Designed to simplify interactions with your deployed smart contracts. These hooks are wrappers around wagmi, an easy-to-use interface with TS autocompletions for reading from, writing to, and monitoring events emitted by your smart contracts". I'm not the best with English tho haha

Also, (maybe for another PR?) I'd have a link to the docs (component section, hook section) "Check out all the available hooks" / "Check out all the components". A button might be too strong, maybe just an underlined link.

**Comment by technophile-04** [comment]
Love all the suggestions, and will implement them tysm 🙌 !!

**Comment by Pabl0cks** [comment]
Awesome job @technophile-04 it's looking great! 🙌

> As you suggested. Move it below "Components":

I like that idea you both proposed, it feels like having all the "main" features or the core from SE-2 above the "_Buidl in Community_" band, and we can add future "secondary" features below that band.

> Also, (maybe for another PR?) I'd have a link to the docs (component section, hook section) "Check out all the available hooks" / "Check out all the components". A button might be too strong, maybe just an underlined link.

I like that idea a lot also, with the underlined links.

> Should we keep it on mobile or remove it?

I feel like maybe in some mobile devices it will feel too much height.  
I'll bring another idea I saw in other websites when showing code snippets on mobile. They make the snippet horizontally scrollable, maybe in our case we could let the comment grow and make the rest scrollable to not show a super big scroll.

**Scrollable example with comment scrollable too:**

scrollable-code-comment-1-line


**Scrollable example with comment not scrollable:**

scrollable-code-comment-growing

**Comment by carletex** [comment]
> Scrollable example with comment scrollable too

As a user, I personally like this one. Thanks for the suggestion Pablo!

**Comment by technophile-04** [comment]
Just updated it with suggested changes, please feel free to nitpick !!

--- 

> Scrollable example with comment scrollable too:

https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/ecd72ac6-fefc-4328-b3a9-bc28a2ba1e98

Just added a `<br/>` at comment so that it breaks into two lines for small deveices and does not go too far 

Tysm Pablo for the great suggestion 🙌 !! 

> Also, (maybe for another PR?) I'd have a link to the docs (component section, hook section) "Check out all the available hooks" / "Check out all the components"

💯💯

File: packages/nextjs/components/HooksExample.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  return <span className="text-[#7B61FF]">{children}</span>;
+};
+export const HooksExample = () => {
+  return (
+    <div className="w-full flex-grow  bg-white rounded-2xl p-6 drop-shadow-[0px_0px_10px_rgba(139,178,241,0.50)]">
+      <div className="bg-base-300/40 w-full h-full border border-primary rounded-3xl p-4 lg:p-6 text-xs lg:text-sm  font-mono overflow-x-scroll whitespace-nowrap  lg:overflow-auto lg:whitespace-normal">
+        <p className="my-3">
+          <Red>import</Red>
+          {" { "} useScaffoldContractWrite {" } "}
+          <Red>from</Red> <Cyan>&quot; ~~/hooks/scaffold-eth &quot;</Cyan>;
```
Comment (Pabl0cks) [inline packages/nextjs/components/HooksExample.tsx:20]: I think there is extra space before and after the path:

```suggestion
          <Red>from</Red> <Cyan>&quot;~~/hooks/scaffold-eth&quot;</Cyan>;
```

File: packages/nextjs/components/HooksExample.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        <p className="my-3">
+          <Red>const</Red>
+          {" { "}writeAsync, isLoading, isMining{" } "}
+          <Cyan> = </Cyan>
+          <br />
+          <Purple>useScaffoldContractWrite</Purple>({"{"}
+          <br />
+          <span className="inline-block ml-4">
+            <Cyan>contractName</Cyan>: <Cyan> &quot;YourContract&quot;</Cyan>,
```
Comment (Pabl0cks) [inline packages/nextjs/components/HooksExample.tsx:31]: One question, should `contractName:`, `functionName:`, `args:` and `blockConfirmation:` have a different color than their values? (like black for example?). 

Maybe is just a bias from my VSC coloring.

File: packages/nextjs/components/HooksExample.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            <br />
+            <Cyan>args</Cyan>: [<Cyan>&quot;The value to set&quot;</Cyan>],
+            <br />
+            <Cyan>blockConfirmation</Cyan>: <Cyan>1</Cyan>,
+            <br />
+            <Purple>onBlockConfirmation</Purple>: (txnReceipt) =&gt; {"{"}
+            <br />
+            <span className="inline-block ml-4">
+              console.<Purple>log</Purple>(<Cyan>&quot;Transaction blockHash&quot;</Cyan>, &nbsp;txnReceipt.
+              <Cyan>blockHash</Cyan>
```
Comment (Pabl0cks) [inline packages/nextjs/components/HooksExample.tsx:43]: Same for this one, I expected `blockHash` to have the same color as `txnReceipt`, but maybe is just what I'm used to.

File: packages/nextjs/pages/index.tsx [hunk 4]
```diff
@@ -179,23 +202,27 @@ const Home: NextPage = () => {
...
-          <div className="space-y-6">
-            <div className="flex items-center justify-center lg:flex-col lg:items-start lg:justify-start gap-2 pt-4 lg:pt-0">
+        <div className="container max-w-[90%] lg:max-w-7xl m-auto py-24 lg:pb-20 lg:pt-28 lg:pl-4  flex flex-col justify-between items-center lg:flex-row lg:items-start gap-5 lg:gap-0">
+          <div className="flex flex-col items-center">
+            <div className="max-w-[400px] lg:max-w-none">
+              <Image src="/assets/se2BlockExplorer.png" alt="Block Explorer" width={1400} height={1400} />
+            </div>
+          </div>
+
+          <div className="space-y-6 flex flex-col items-center lg:pl-16 lg:pt-4">
```
Comment (Pabl0cks) [inline packages/nextjs/pages/index.tsx:212]: First time I have opened this last version, the "*Block Explorer*" section felt a bit too close to "*Buidl in Community*" band.

Checking old version with previous zig-zag, it seemed that the text content was a bit farther from the band, maybe we could add a bit more of `pt` on `lg` devices, at least to that text content? Maybe just a personal perception tbh 😅.

Previous:

image

**Review by Pabl0cks — CHANGES_REQUESTED** [review]

**Comment by Pabl0cks** [comment]
> Just updated it with suggested changes, please feel free to nitpick !!

GJ Shiv, it looks really good for me 🙌. 

I've added some nitpicks, probably most of them are subjective perceptions, so feel free to ignore them and close 😊

File: packages/nextjs/components/HooksExample.tsx (hunk 1 above)
Comment (technophile-04) [reply packages/nextjs/components/HooksExample.tsx]: Yup 🙌

File: packages/nextjs/components/HooksExample.tsx (hunk 2 above)
Comment (technophile-04) [reply packages/nextjs/components/HooksExample.tsx]: Yup, actually Figma had the values in a bit darker : 

Screenshot 2023-10-18 at 12 44 21 PM

I updated it the same as Figma, like it differentiates them a bit but also looks simple and clean

Screenshot 2023-10-18 at 12 53 31 PM

even [tokyonight theme](https://github.com/folke/tokyonight.nvim) does it similarly were keys are dark and values are light

---

I don't know if having a different color will make it more overpowered, this is how it looks with black : 

Screenshot 2023-10-18 at 11 45 46 AM

**Comment by technophile-04** [comment]
Tysm Pablo!! Just added the changes 🙌

---

Also one last thing what if change title from "Custom React hooks and utils" => "Custom Wagmi hooks and utils"  ? 

Reason: 

- Like currently wagmi is kinda of hidden and don't want people to get the wrong notion that we are still using [eth-hooks](https://github.com/scaffold-eth/eth-ui) and seeing `wagmi` people might feel at home (not another extra thing to learn to use Scaffold-eth)

- Also this https://x.com/awkweb/status/1692562844619817297?s=20

File: packages/nextjs/components/HooksExample.tsx (hunk 2 above)
Comment (Pabl0cks) [reply packages/nextjs/components/HooksExample.tsx]: I think is cool as you've set it 🙌

**Comment by Pabl0cks** [comment]
> Tysm Pablo!! Just added the changes 🙌

TY Shiv, great job! I see this ready to ship 🚀

I don't have enough context for the wagmi title, looks good to me, so I'll let Carlos decide ♥

**Comment by carletex** [comment]
This is looking great @technophile-04 

Let's merge it and we can always tweak.

> Also one last thing what if change title from "Custom React hooks and utils" => "Custom Wagmi hooks and utils" ?

I like this!

Pushed it 19d582f5fc6d4363c917f9d044ac773f62525f8c

**Pull request merged**